zio-http/src/main/scala/zhttp/http/HttpData.scala

```diff
@@ -29,25 +33,74 @@ sealed trait HttpData { self =>
     case _              => false
   }
 
-  def toByteBuf: Task[ByteBuf] = {
+  final def toByteBuf: Task[ByteBuf] = {
     self match {
-      case HttpData.Text(text, charset)   => UIO(Unpooled.copiedBuffer(text, charset))
-      case HttpData.BinaryChunk(data)     => UIO(Unpooled.copiedBuffer(data.toArray))
-      case HttpData.BinaryByteBuf(data)   => UIO(data)
-      case HttpData.Empty                 => UIO(Unpooled.EMPTY_BUFFER)
-      case HttpData.BinaryStream(stream)  =>
-        stream
-          .asInstanceOf[ZStream[Any, Throwable, ByteBuf]]
-          .fold(Unpooled.compositeBuffer())((c, b) => c.addComponent(b))
-      case HttpData.RandomAccessFile(raf) =>
-        effectBlocking {
-          val fis                      = new FileInputStream(raf().getFD)
-          val fileContent: Array[Byte] = new Array[Byte](raf().length().toInt)
-          fis.read(fileContent)
-          Unpooled.copiedBuffer(fileContent)
+      case HttpData.Incoming(unsafeRun) =>
+        for {
+          buffer <- UIO(Unpooled.compositeBuffer())
+          body   <- ZIO.effectAsync[Any, Throwable, ByteBuf](cb =>
+            unsafeRun(ch =>
+              msg => {
+                buffer.writeBytes(msg.content.content())
+                if (msg.isLast) {
+                  cb(UIO(buffer) ensuring UIO(ch.ctx.pipeline().remove(HTTP_CONTENT_HANDLER)))
```

Comment:
This is too much power IMO. One should not be able to change the pipeline from here. 
Modifying the pipeline should only be done in `Handler.scala`